packages/workspace/src/browser/workspace-service.ts

```diff
@@ -77,10 +79,25 @@ export class WorkspaceService implements FrontendApplicationContribution {
      */
     onStop(app: FrontendApplication): void {
         if (this._root) {
-            this.server.setRoot(this._root.uri);
+            this.server.setWorkspace(this._root.uri);
+        }
+    }
+
+    async onStart() {
+        const allWorkspace = await this.recentWorkspaces();
+        if (allWorkspace.length > 0) {
+            this.hasWorkspace = true;
         }
     }
 
+    get hasAWorkspace(): boolean {
```

Comment:
I am not sure about this wording. It might be confused with having a workspace currently set or not? Maybe hasHistory() instead?

Reply:
Also I wonder if this is needed. Clients of this service could instead call recentWorkspaces() and if they get an empty list, they know there is no workspace history

Reply:
With this call, we check if there is a workspace or not already defined in recentworkspace and set the sensitivity of the buttons under the file menu.  
I will rename it as suggested "hasHistory()